File: cpp/part1.cpp
#include<iostream>
#include<cstdlib>
#define Max 100
using namespace std;

int main () {
  // int nums[] = {3,2,1,4};int len = sizeof(nums)/sizeof(nums[0]);
  int nums[Max];
  int number;
  cout << "enter a number between 100 : " << endl;
  cin >> number;

  for (int i=0;i<number;i++) {
    nums[i] = rand();
  }

  for (int i=0; i<number ;i++){
    for (int j=i+1;j<number;j++){
      if (nums[i] > nums[j]) {
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
      }
    }
  }

  for (int i=0;i<number;i++){
    cout << nums[i] << " ";
  }
  cout << endl;
  // time complexity O(N^2) && space complexity O(1)
  return 0;
}

File: js/part1.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const p = new Promise((resolve, error) => {
  rl.question("enter a number", (x) => {
    if (x < 0 || x > 100) error("not a valid input");
    resolve(x);
  });
  rl.close();
});

rl.question("enter a number", (x) => {
  console.log(x);
});

const nums = [];
let num = Math.floor(Math.random() * 100);

for (let i = 0; i < num; i++) {
  nums.push(Math.floor(Math.random() * 100000));
}

for (let i = 0; i < nums.length; i++) {
  for (let j = i + 1; j < nums.length; j++) {
    if (nums[i] > nums[j]) {
      let temp = nums[i];
      nums[i] = nums[j];
      nums[j] = temp;
    }
  }
}

// time complexity O(N^2) && space complexity O(1)
console.log(nums);
